conans/client/build/cmake_flags.py

```diff
@@ -263,11 +251,21 @@ def get_definitions(self):
         if compiler_version:
             ret["CONAN_COMPILER_VERSION"] = str(compiler_version)
 
-        # Force compiler flags -- TODO: give as environment/setting parameter?
+        # C, CXX, LINK FLAGS
         arch_flag = architecture_flag(compiler=compiler, arch=arch)
-        ret = add_cmake_flag(ret, 'CONAN_CXX_FLAGS', arch_flag)
-        ret = add_cmake_flag(ret, 'CONAN_SHARED_LINKER_FLAGS', arch_flag)
-        ret = add_cmake_flag(ret, 'CONAN_C_FLAGS', arch_flag)
+        if arch_flag:
+            ret['CONAN_CXX_FLAGS'] = arch_flag
+            ret['CONAN_SHARED_LINKER_FLAGS'] = arch_flag
+            ret['CONAN_C_FLAGS'] = arch_flag
+            if self._set_cmake_flags:
+                ret['CMAKE_CXX_FLAGS'] = arch_flag
+                ret['CMAKE_SHARED_LINKER_FLAGS'] = arch_flag
+                ret['CMAKE_C_FLAGS'] = arch_flag
+        elif self._parallel and "Windows" in str(os_) and compiler == "Visual Studio":
+            # Parallel and Windows is mutually exclusive from above "arch_flag"
```

Comment:
why? I don't see how is related the `arch_flag` with the parallel

Reply:
They are mutually exclusive. The arch_flag is only set for gcc, clang, so it is impossible that CONAN_CXX_FLAGS is set both with flags from ``arch_flag`` and parallel VS flags. This is the reason that bug was never fired, it is unreachable.

I could leave it and create a simple unittest instead.

Reply:
The thing is that it is not transparent nor this function responsibility (as it is written) to know that "arch_flag" is not adjusted for visual studio, that's why the code looks wrong. I would prefer to call `architecture_flag` only when not windows or something that helps to understand.